frontend/src/js/components/tenants/TenantCreateForm.tsx

```diff
@@ -145,15 +145,22 @@ export const TenantCreateForm = (props: TenantCreateFormProps) => {
     dispatch(getSsoConfigs());
   }, [dispatch]);
 
-  const submitNewTenant = async data => {
-    const { email, password, device_limit, send_reset_password, ...remainder } = data;
-    if (adminExists) {
-      await dispatch(addTenant({ users: [{ role: rolesByName.admin, email }], device_limit: Number(device_limit), ...remainder }));
-    } else {
-      await dispatch(addTenant({ admin: { password, email, send_reset_password }, device_limit: Number(device_limit), ...remainder }));
-    }
-    onCloseClick();
-  };
+  const submitNewTenant = useCallback(
+    async data => {
+      const { email, password, device_limit, send_reset_password, ...remainder } = data;
+      let selectionState = { device_limit: Number(device_limit), ...remainder };
+      if (adminExists) {
+        selectionState = { users: [{ role: rolesByName.admin, email }], ...selectionState };
+      } else {
+        selectionState = { admin: { password, email, send_reset_password }, ...selectionState };
+      }
+      const success = await dispatch(addTenant(selectionState)).unwrap();
+      if (success) {
```

Comment:
`if (success){}` is not necessary, as `.unwrap()` will throw an exception in case of rejection
https://redux-toolkit.js.org/api/createAsyncThunk#checking-errors-after-dispatching

Reply:
Thanks for the heads up! With the experience from the RTK migration where the unwrapping was missed when it wasn't bound to an assignment, I would rather be explicit - but with the recent change in the UX design the handling is no longer optional 👌